src/data/complex/exponential.lean

```diff
@@ -394,6 +394,8 @@ by rw lim_mul_lim;
   exact eq.symm (lim_eq_lim_of_equiv (by dsimp; simp only [hj];
     exact cauchy_product (is_cau_abs_exp x) (is_cau_exp y)))
 
+attribute [irreducible] complex.exp
+

```

Comment:
I don't think the lemmas about `complex.exp` uniquely identify it as such yet. I think all the lemmas I proved are true of `(e+0.000001)^x`.

Reply:
I think we can still add the attribute. After this attribute people should either be proving things using your (partial) API, or the should extend the API. This extending can either happen in this file, before the attribute is changed, or possibly in another file where they locally mark `complex.exp` as reducible again.

Reply:
We can add to the API a lemma `exp_def` which expands `exp` (or `exp z`) to its definition.